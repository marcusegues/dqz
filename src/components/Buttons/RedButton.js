import React from 'react';
import { ScrollView, Text, View } from 'react-native';
import Touchable from 'react-native-platform-touchable';
import * as colors from './../../styles/colors';

const RedButton = ({ text, onPress, style }) => (
  <View>
    {/*<View style={{ margin: 16 }}>*/}
    <Touchable
      onPress={onPress}
      style={{ backgroundColor: colors.MAIN_RED, ...style }}
      background={Touchable.Ripple(colors.MAIN_RED)}
    >
      <Text style={{ textAlign: 'center', color: '#fff' }}>{text}</Text>
    </Touchable>
  </View>
);
export default RedButton;
