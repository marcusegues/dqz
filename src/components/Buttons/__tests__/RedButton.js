import renderer from 'react-test-renderer';
import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';
import Touchable from 'react-native-platform-touchable';
import { RedButton } from '../RedButton';

describe('RedButton', () => {
  test('renders according to snapshot', () => {
    const component = renderer
      .create(<RedButton text="text" onPress={jest.fn()} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('has uppercase text', () => {
    const str = 'foo';
    const component = shallow(<RedButton text={str} onPress={jest.fn()} />);
    expect(component.containsMatchingElement(Text)).toBe(true);
    expect(
      component.containsMatchingElement(<Text>{str.toUpperCase()}</Text>)
    ).toBe(true);
  });

  test('we can click it', () => {
    const clickFn = jest.fn();
    const component = shallow(<RedButton text="text" onPress={clickFn} />);
    component.find(Touchable).forEach(child => {
      child.simulate('press');
    });
    expect(clickFn).toBeCalled();
  });
});
