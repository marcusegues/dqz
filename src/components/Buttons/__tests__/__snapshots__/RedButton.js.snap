// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedButton renders according to snapshot 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignSelf": "center",
      "backgroundColor": "#DC0018",
      "borderColor": "#ddd",
      "borderRadius": 3,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 31.388235294117646,
      "width": "95%",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#DC0018",
        "borderRadius": 3,
        "flex": 1,
        "opacity": 1,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 3,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontFamily": "roboto_medium",
          "fontSize": 22,
          "fontWeight": "500",
          "paddingVertical": 31.388235294117646,
        }
      }
    >
      TEXT
    </Text>
  </View>
</View>
`;

exports[`RedButton renders according to snapshot if disabled 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignSelf": "center",
      "backgroundColor": "#DC0018",
      "borderColor": "#ddd",
      "borderRadius": 3,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginTop": 31.388235294117646,
      "width": "95%",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "lightgray",
        "borderRadius": 3,
        "flex": 1,
        "opacity": 1,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 3,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontFamily": "roboto_medium",
          "fontSize": 22,
          "fontWeight": "500",
          "paddingVertical": 31.388235294117646,
        }
      }
    >
      TEXT
    </Text>
  </View>
</View>
`;
